x-pack/plugins/observability/public/pages/rule_details/index.tsx

```diff
@@ -536,22 +532,20 @@ export function RuleDetailsPage() {
           onSave: reloadRule,
         })}
       <DeleteModalConfirmation
-        onDeleted={async () => {
+        onDeleted={() => {
           setRuleToDelete([]);
           navigateToUrl(http.basePath.prepend(paths.observability.rules));
         }}
-        onErrors={async () => {
+        onErrors={() => {
           setRuleToDelete([]);
           navigateToUrl(http.basePath.prepend(paths.observability.rules));
         }}
-        onCancel={() => {}}
+        onCancel={() => setRuleToDelete([])}
         apiDeleteCall={deleteRules}
         idsToDelete={ruleToDelete}
         singleTitle={rule.name}
         multipleTitle={rule.name}
-        setIsLoadingState={(isLoading: boolean) => {
-          setIsPageLoading(isLoading);
-        }}
+        setIsLoadingState={() => setIsPageLoading(true)}
```

Comment:
https://github.com/elastic/kibana/blob/main/x-pack/plugins/triggers_actions_ui/public/application/sections/rule_details/components/rule_details.tsx#L288

Reply:
We need to tell the user once he clicks the delete button that there is something in the background happening (deleting) instead of nothing.